File: src/ecs/components.py
from dataclasses import dataclass


@dataclass
class Camera:
    pass


@dataclass
class Player:
    pass


@dataclass
class Position:
    x: float
    y: float


@dataclass
class Velocity:
    x: float
    y: float


@dataclass
class Display:
    color: str | tuple

File: rustext/src/ecs/components.rs
use godot::prelude::*;

#[derive(Clone, Debug)]
pub struct Display {
    pub sprite_id: InstanceId,
    pub base_color: Color,
}
pub struct Player;
pub struct Position(pub Vector2i);
pub struct Velocity(pub Vector2i);
pub struct SyncSpritePosition;
pub struct Despawn;
